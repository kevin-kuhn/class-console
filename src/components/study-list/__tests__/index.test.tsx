import { render } from '../../../test/test-utils'
import StudyList from '../index'

import '@testing-library/jest-dom'

describe('StudyList', () => {
  it('should be in document', () => {
    const { container } = render(
      <StudyList items={[]} onChangeStudy={() => {}} />
    )

    expect(container).toBeInTheDocument()
  })

  it('should match snapshot when empty', () => {
    const { container } = render(
      <StudyList items={[]} onChangeStudy={() => {}} />
    )

    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when full', () => {
    const { container } = render(
      <StudyList
        items={[
          {
            type: 'video',
            title: 'Introdução à Geografia',
            slug: 'introducao-a-geografia'
          },
          {
            type: 'video',
            title: 'Coordenadas geográficas e fuso-horário',
            slug: 'coordenadas-geograficas-e-fusohorario'
          },
          {
            type: 'text',
            title: 'Material de apoio',
            slug: 'material-de-apoio'
          },
          {
            type: 'text',
            title:
              'Embate entre igualdade e globalização na sociedade contemporânea',
            slug: 'proposta-de-redacao'
          },
          {
            type: 'exercise',
            title: 'Exercício 1',
            slug: 'exercicio-1'
          },
          {
            type: 'exercise',
            title: 'Exercício 2',
            slug: 'exercicio-2'
          }
        ]}
        onChangeStudy={() => {}}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
