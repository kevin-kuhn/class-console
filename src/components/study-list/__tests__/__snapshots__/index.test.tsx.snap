// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyList should match snapshot when empty 1`] = `
<div>
  <ul
    class="ul"
  />
</div>
`;

exports[`StudyList should match snapshot when full 1`] = `
<div>
  <ul
    class="ul"
  >
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 10l4.553-2.276A1 1 0 0121 8.618v6.764a1 1 0 01-1.447.894L15 14M5 18h8a2 2 0 002-2V8a2 2 0 00-2-2H5a2 2 0 00-2 2v8a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Introdução à Geografia
        </span>
      </p>
    </li>
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 10l4.553-2.276A1 1 0 0121 8.618v6.764a1 1 0 01-1.447.894L15 14M5 18h8a2 2 0 002-2V8a2 2 0 00-2-2H5a2 2 0 00-2 2v8a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Coordenadas geográficas e fuso-horário
        </span>
      </p>
    </li>
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Material de apoio
        </span>
      </p>
    </li>
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Embate entre igualdade e globalização na sociedade contemporânea
        </span>
      </p>
    </li>
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Exercício 1
        </span>
      </p>
    </li>
    <li
      class="li "
    >
      <p
        class="p"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <span>
          Exercício 2
        </span>
      </p>
    </li>
  </ul>
</div>
`;
